apps/server/src/modules/files-storage/client/s3-client.adapter.spec.ts

```diff
@@ -46,136 +51,225 @@ describe('S3ClientAdapter', () => {
+				client.send.mockRejectedValueOnce(error);
 
-		it('should return file', async () => {
-			const result = await service.get(pathToFile);
-			expect(result).toStrictEqual(
-				expect.objectContaining({
-					contentType: 'data.ContentType',
-					contentLength: 'data.ContentLength',
-					contentRange: 'data.ContentRange',
-					etag: 'data.ETag',
-				})
-			);
-		});
+				return { error, pathToFile };
+			};
 
-		it('should throw error from s3 client if NoSuchKey', async () => {
-			const e = new Error('NoSuchKey');
-			client.send = jest.fn().mockRejectedValue(e);
+			it('should throw NotFoundException', async () => {
+				const { pathToFile } = setup('NoSuchKey');
 
-			await expect(service.get(pathToFile)).rejects.toThrowError(new NotFoundException('NoSuchKey'));
-		});
+				await expect(service.get(pathToFile)).rejects.toThrowError(NotFoundException);
+			});
 
-		it('should throw error from client', async () => {
-			const e = new Error('Bad Request');
-			client.send = jest.fn().mockRejectedValue(e);
+			it('should throw error', async () => {
+				const { pathToFile } = setup('Unknown Error');
 
-			await expect(service.get(pathToFile)).rejects.toThrowError();
+				await expect(service.get(pathToFile)).rejects.toThrowError(InternalServerErrorException);
+			});
 		});
 	});
 
 	describe('create', () => {
-		const readable = Readable.from('ddd');
-		const file = new FileDto({
-			data: readable,
-			name: 'test.txt',
-			mimeType: 'text/plain',
-		});
-		const path = 'test/test.txt';
+		const createFile = () => {
+			const readable = Readable.from('ddd');
+			const file = new FileDto({
+				data: readable,
+				name: 'test.txt',
+				mimeType: 'text/plain',
+			});
+
+			return { file };
+		};
 
-		it('should return data', async () => {
-			client.send = jest.fn().mockResolvedValue(true);
+		describe('WHEN file is created successfully', () => {
+			const setup = () => {
+				const { file } = createFile();
+				const { pathToFile } = createParameter();
 
-			const result = await service.create(path, file);
+				client.config.endpoint = jest.fn().mockResolvedValueOnce({ protocol: '' });
 
-			expect(result).toBeDefined();
-		});
+				return { file, pathToFile };
+			};
 
-		it('should throw error from client', async () => {
-			const e = new Error('Bad Request');
+			it('should return data', async () => {
+				const { file, pathToFile } = setup();
 
-			client.send = jest.fn().mockRejectedValue(e);
+				const result = await service.create(pathToFile, file);
 
-			await expect(service.create(path, file)).rejects.toThrow();
+				expect(result).toBeDefined();
+			});
 		});
 
-		it('should call createBucket() if error from client has error.Code === "NoSuchBucket" ', async () => {
-			const e = { Code: 'NoSuchBucket' };
-			client.send = jest.fn().mockRejectedValueOnce(e);
-			service.createBucket = jest.fn();
-			await service.create(path, file);
-			expect(service.createBucket).toBeCalled();
+		describe('WHEN client throws error', () => {
```

Comment:
I would still like it better if this test would be below the next. Then it would be easier to understand if you read from the top, I think, because you have the special case first.